src/ConnectedMode/Persistence/ServerConnectionsRepository.cs

```diff
@@ -0,0 +1,236 @@
+        }
+        catch (Exception ex) when (!ErrorHandler.IsCriticalException(ex))
+        {
+            logger.WriteLine(ex.Message);
+        }
+        return false;
+    }
+
+    private bool TryAddConnection(List<ServerConnection> connections, ServerConnection connection)
+    {
+        if (connection.Credentials is null)
+        {
+            logger.LogVerbose($"Connection was not added.{nameof(ServerConnection.Credentials)} is not filled");
+            return false;
+        }
+
+        if (connections.Find(x => x.Id == connection.Id) is not null)
+        {
+            logger.LogVerbose($"Connection was not added.{nameof(ServerConnection.Id)} already exist");
+            return false;
+        }
+
+        try
+        {
+            connections.Add(connection);
+            credentialsLoader.Save(connection.Credentials, connection.CredentialsUri);
+            return true;
+        }
+        catch (Exception ex) when (!ErrorHandler.IsCriticalException(ex))
+        {
+            logger.WriteLine(ex.Message);
+        }
+
+        return false;
+    }
+
+
+    private void TryDeleteCredentials(ServerConnection removedConnection)
+    {
+        try
+        {
+            credentialsLoader.DeleteCredentials(removedConnection.CredentialsUri);
+        }
+        catch (Exception ex) when (!ErrorHandler.IsCriticalException(ex))
+        {
+            logger.WriteLine(ex.Message);
+        }
+    }
+
+    private static bool TryUpdateConnectionSettings(List<ServerConnection> connections, string connectionId, ServerConnectionSettings connectionSettings)
+    {
+        var serverConnectionToUpdate = connections?.Find(c => c.Id == connectionId);
+        if (serverConnectionToUpdate == null)
+        {
+            return false;
+        }
+
+        serverConnectionToUpdate.Settings = connectionSettings;
+        return true;
+    }
+
+    private static string GetStorageFilePath(IEnvironmentVariableProvider environmentVariables)
+    {
+        var appDataFolder = environmentVariables.GetSLVSAppDataRootPath();
+        return Path.Combine(appDataFolder, ConnectionsFileName);
+    }
+
+    private List<ServerConnection> ReadServerConnectionsFromFile()
+    {
+        try
+        {
+            var model = jsonFileHandle.ReadFile<ServerConnectionsListJsonModel>(storageFilePath);
+            return model.ServerConnections.Select(serverConnectionModelMapper.GetServerConnection).ToList();
+        }
+        catch (FileNotFoundException)
+        {
+            // file not existing should not be treated as an error, as it will be created at the first write
+            return [];
+        }
+        catch(Exception ex) when (!ErrorHandler.IsCriticalException(ex))
+        {
+            logger.WriteLine(ex.Message);
+        }
+
+        return null;
+    }
+
+    private bool SafeUpdateConnectionsFile(Func<List<ServerConnection>, bool> tryUpdateConnectionModels)
+    {
+        lock (LockObject)
+        {
+            try
```

Comment:
I don't think try catch is needed here. Read... already does logging, mapper should not throw normally and TryWrite does not throw

Reply:
It's exactly the mapper that can throw, more exactly: GetServerConnectionsListJsonModel and, more specifically, the GetServerConnectionJsonModel